Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabUndo.cpp

```diff
@@ -261,8 +261,13 @@ namespace AzToolsFramework
             instanceDom.CopyFrom(instanceDomRef->get(), instanceDom.GetAllocator());
 
             //apply the patch to the template within the target
-            AZ::JsonSerializationResult::ResultCode result = AZ::JsonSerialization::ApplyPatch(instanceDom,
-                instanceDom.GetAllocator(), patch, AZ::JsonMergeApproach::JsonPatch);
+            AZ::JsonSerializationResult::ResultCode result = PrefabDomUtils::ApplyPatches(instanceDom, instanceDom.GetAllocator(), patch);
+
+            AZ_Error(
+                "Prefab",
+                result.GetOutcome() != AZ::JsonSerializationResult::Outcomes::PartialSkip &&
+                    result.GetOutcome() != AZ::JsonSerializationResult::Outcomes::Success,
```

Comment:
Nitpick: this logic is hard to read. Consider the simpler:
```
result.GetOutcome() == AZ::JsonSerializationResult::Outcomes::PartialSkip ||
result.GetOutcome() == AZ::JsonSerializationResult::Outcomes::Success
```

Reply:
will change